### ISSUE TITLE
Support DCRDEX seed format and use it as default

### ISSUE THREAD

**Issue by jzbz** [decred/decrediton#3940]
DCRDEX has changed to a custom seed format that is considerably more user friendly than the one we are currently using (https://github.com/decred/dcrdex/pull/2710). We should ideally harmonize all Decred software to be compatible and it may make sense to adopt this format as the default for Decrediton in addition to supporting restoring wallets that use this format.

Advantages:
- Shorter.
- BIP-39 word list is supported by all seed backup products.
- Doesn't make use of the BIP-39 standard (only words).
- Embedded birthday to make wallet restorations faster.

Code: https://github.com/decred/dcrdex/tree/master/client/mnemonic

**Comment by JoeGruffins** [comment]
Does dcrwallet allow restoring from a date currently? If not should we add that?

**Comment by jzbz** [comment]
> Does dcrwallet allow restoring from a date currently? If not should we add that?

I don't believe it currently supports specifying a restore height, whether dcrwallet should have that feature is above my head, maybe @jrick can chime in on how best to approach this.

**Comment by jrick** [comment]
not currently supported but the feature would be very welcome.  a birthday height can be provided either in the initial prompts or during wallet creation with an added config option.  prompting for it would require saving this data to the db so it can be used later on, but we would need to do that anyways if the seed format encoded a birthday.
